From 3e8afd1896f4f6dfecc2bf9658f631409348940d Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Fri, 29 Jul 2022 10:06:15 +0800
Subject: [PATCH 3656/4585] [Huawei] mm: Make shmem reliable's two limit
 consistent
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187270

--------------------------------

The default two limits of reliable shmem are not consistent. The default
value of shmem_reliable_nr_page is LONG_MAX and
sysctl_shmem_reliable_bytes_limit is ULONG_MAX, these two value will be
consistent if updated. Since no system can have such huge memory, no real
problem will be produced. But this will do produce user confusion.

To fix this confusion, set the default value of shmem_reliable_nr_page to
ULONG_MAX >> PAGE_SHIFT.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
---
 mm/mem_reliable.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/mem_reliable.c b/mm/mem_reliable.c
index 278e40199e66..7a3ea31e58a3 100644
--- a/mm/mem_reliable.c
+++ b/mm/mem_reliable.c
@@ -32,7 +32,7 @@ struct percpu_counter anon_reliable_pages;
 static unsigned long reliable_pagecache_max_bytes = ULONG_MAX;
 /* reliable user limit for user tasks with reliable flag */
 unsigned long task_reliable_limit = ULONG_MAX;
-long shmem_reliable_nr_page = LONG_MAX;
+long shmem_reliable_nr_page = ULONG_MAX >> PAGE_SHIFT;
 atomic_long_t reliable_user_used_nr_page;
 
 bool mem_reliable_counter_initialized(void)
-- 
2.17.1

